x/ccv/provider/keeper/power_shaping.go

```diff
@@ -111,6 +111,54 @@ func (k Keeper) CapValidatorSet(
 	}
 }
 
+// CapValidatorSet caps the provided `validators` if chain with `consumerId` is an Opt In chain with a validator-set cap.
+// It prioritizes validators from the priority list and then adds remaining validators up to the cap.
+func (k Keeper) CapValidatorSet(
+	ctx sdk.Context,
+	powerShapingParameters types.PowerShapingParameters,
+	validators []types.ConsensusValidator,
+) []types.ConsensusValidator {
+	if powerShapingParameters.Top_N > 0 {
+		// is a no-op if the chain is a Top N chain
+		return validators
+	}
+
+	validatorSetCap := powerShapingParameters.ValidatorSetCap
+	if validatorSetCap == 0 || int(validatorSetCap) >= len(validators) {
+		return validators
+	}
+
+	// Filter and sort the priority list
```

Comment:
maybe to be more specific

```suggestion
	// Filter and sort the priority list by power
```